mypy/main.py

```diff
@@ -31,6 +33,40 @@ def __init__(self) -> None:
         self.dirty_stubs = False
 
 
+class MyPyCommand(setuptools.Command):
+    """The :class:`MyPyCommand` class is used by setuptools to perform
+    type checks on registered modules.
+    """
+
+    description = "Validate PEP 0484 Types with mypy"
+    user_options = [
+        ('mypy-args=', None, 'Arguments to pass through to mypy')
+    ]  # type: List[tuple]
+
+    def initialize_options(self):
+        self.mypy_args = ''
+
+    def finalize_options(self):
+        pass
+
+    def run(self):
+        try:
+            mypy_options = []
+            for package in self.distribution.packages:
+                mypy_options.append('--package')
+                mypy_options.append(package)
+            mypy_options.extend(self.mypy_args.split())
```

Comment:
Totally agree that the command line story here is kind of janky (None of it is very well documented).  It was the least bad option I could come up with without also totally refactoring how we parse options.

`setuptools` use `fancy_getopt` (their own option parser found here - http://svn.python.org/projects/python/tags/r32rc1/Lib/distutils/fancy_getopt.py).  It doesn't appear that it supports `argparse` style `nargs`

`flake8` uses a clever hack to convert their OptParser to getopt - https://gitlab.com/pycqa/flake8/blob/master/flake8/main.py#L90-97 .  If we migrated argument parsing to argparse we could do the same.

I updated this to use `shlex`, which will respect quotes for splitting.  

`--mypy-arg='foo bar --raz "hello world"'`

Will work